docs/spec/lcd/api.md

```diff
@@ -0,0 +1,847 @@
+
+url: /keys/{name}, Method: GET
+
+Functionality: Get key information according to the specified key name
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+    	"name": "test",
+          "address": "cosmosaccaddr1thlqhjqw78zvcy0ua4ldj9gnazqzavyw4eske2",
+          "pub_key": "cosmosaccpub1zcjduc3qyx6hlf825jcnj39adpkaxjer95q7yvy25yhfj3dmqy2ctev0rxmse9cuak"
+    },
+    "rest api": "2.0"
+}
+
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot find corresponding name",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /keys/{name} - PUT
+
+url: /keys/{name}, Method: PUT
+
+Functionality: Update key password
+
+Parameters: 
+
+| Parameter       | Type   | Default | Required | Description     |
+| --------------- | ------ | ------- | -------- | --------------- |
+| old_password    | string | null    | true     | password before |
+| new_password    | string | null    | true     | password before |
+| repeat_password | string | null    | true     | password before |
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "updated":name
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot update the corresponding key",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /keys/{name} - DELETE
+
+url: /keys/{name}, Method: DELETE
+
+Functionality: Delete key from keystore
+
+Parameters:
+
+| Parameter | Type   | Default | Required | Description      |
+| --------- | ------ | ------- | -------- | ---------------- |
+| password  | string | null    | true     | password of keys |
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "deleted":name
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot delete the corresponding key",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+## ICS20 - TokenAPI
+
+The TokenAPI exposes all functionality needed to query account balances and send transactions.
+
+### /balance/{account} - GET
+
+url: /balance/{account}, Method: GET
+
+Functionality:
+
+Parameters:
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     {
+         "atom": 1000,
+         "photon": 500,
+         "ether": 20
+     }
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "Invalid account",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /create_transfer - POST
+
+url: /create_transfer, Method: **POST**
```

Comment:
How about making this consistent with the rest of the API and more REST-ful? Like sending y Atom tokens to accountX would be POST `/accounts/accountx/tokens/atoms` {amount: y, name: accountName, password: accountPassword}.
And POST-ing to get sth. back feels weird. Maybe do a GET with some suffix? i.e.: GET `/accounts/accountx/tokens/atoms?build`

Reply:
I don't think we should interleave the TokenAPI with the KeyAPI. The former should have no notion of how to create valid signatures. This will allow the apis to evolve independently from each other and won't require the user to store the keys within the LCD.

Reply:
I would go with both approaches, as I heard from different people already, that they actually like the build -> sign -send way of handling the LCD. I am sure we can find a clever way of implementing this so the parts do not have too many dependencies. Let's get feedback from the users and decide if we want to drop the bss way of handling things later.

Another construction endpoint idea that feels a little more like a GET: (brainstorming)
GET `/accounts/{address}/tokens/{denomination}/transfer_tx` {from, amount}